todoCLI.js

```diff
@@ -0,0 +1,126 @@
+    input: process.stdin, // gives single input readable stream
+    output: process.stdout, // gives single output writable stream
+    // in node >.load todoCLI.js it will prevent double typing h -> hh(x)
+});
+
+const toDoMenu = (options) => {
+    const caseIgnoreOptions = options.toLowerCase(); // incase of user input is in capital letter
+    // (v) View
+    // View function will show the List which contains all the todos
+    // if list is empty it will mention user that the list is empty
+    // if list isn't empty it will show the list in format:
+    // index [ ] title
+    // if completed: true will be [✓] if false will be [ ]
+    if (caseIgnoreOptions === "v") {
+        console.log("") // added to make \n btw
+        if (myTodos.length === 0) {
+            console.log("\nList is empty...");
+            // to comeout from the rl.question added toDoMenu as a callback
+            rl.question("\n(v) View • (n) New • (cX) Complete • (dX) Delete • (s) Save • (q) Quit\n>", toDoMenu);
+        } else { // if list is not empty
+            for (let i = 0; i < myTodos.length; i++) { // for all element in myTodos.json array(loaded as myTodos)
+                if (myTodos[i].completed === true) { // if completed: true
+                    console.log(`${i} [✓] ${myTodos[i].title}`);
+                } else { // if completed: false
+                    console.log(`${i} [ ] ${myTodos[i].title}`);
+                }
+            }
+        }
+        rl.question("\n(v) View • (n) New • (cX) Complete • (dX) Delete • (s) Save • (q) Quit\n>", toDoMenu);
+
+        // (n) New
+        // New function will add a new todo to the List
+        // will be added as an object in the array
+        // myTodos = [{completed: , title: }]
+    } else if (caseIgnoreOptions === "n") {
+        rl.question("\nWhat\n>", todo => {
+            myTodos.push({
+                completed: false, // when add todo its added as incompleted
+                title: todo // User input for What? will go into here
+            });
+            rl.question("\n(v) View • (n) New • (cX) Complete • (dX) Delete • (s) Save • (q) Quit\n>", toDoMenu);
+        });
+
+        // (cX) Complete
+        // Complete will change completed: false => completed: true
+        // if it is already marked as completed: true will change to completed: false which is incomplete
+        // to receive input cX(c + number of index from the List)
+        // caseIgnoreOptions.slice = "X"
+        // parseInt(caseIgnoreOptions.slice(1)) = X
+        // if X is only composed with integer "X" == X
+        // if not parseInt value will be different and send it to invalid input statement
+    } else if (caseIgnoreOptions.length > 1 && caseIgnoreOptions[0] === "c" && caseIgnoreOptions.slice(1) == parseInt(caseIgnoreOptions.slice(1))) {
+        const X = parseInt(caseIgnoreOptions.slice(1));
+        if (X <= myTodos.length && myTodos[X].completed === false) { // if incompleted
```

Comment:
If the last index of the list is 2 and we try to complete a task that is 1 + the last index e.g. 3 then the program will crash. To fix this we'd need to change the first comparison check to `X < myTodos.length` instead of `<=`.